### PR TITLE
refactor(Select): use portalled dropdown

### LINKED ISSUE #949: [Select] needs to portal to `document.body`
### Component/Module

`Select` in a `Dialog`

### Expected Result

The `Select` dropdown overflows the `Dialog` boundaries

### Actual Result

The `Select` dropdown is rendered within the content area of the `Dialog` which is set to `overflow: scroll`

### PR DESCRIPTION
**Pull request by akdetrick**
closes #949 

- Replaces focus trap library with `react-focus-lock`, which is a better implementation that can follow portals
- Adds "active' state for dropdown trigger used in `Select`
- Replace static CSS positioning in `Select` with the `react-laag` positioning hook (also used in `Popover`)

### The `Select` now portals it's dropdown, allowing the menu control to overflow the `Dialog` container:
select-in-dialog

### The positioning engine (`react-laag`) now automatically flips between top/bottom to avoid getting cut off by viewport edges
select-flip

### CODE REVIEW

File: src/Select/index.js [hunk 1]
```diff
@@ -66,7 +67,9 @@ const Select = ({
   testId,
 }) => {
   // The menu should only render children that have `value` or `onSelect` prop
-  const items = React.Children.toArray(children).filter(({ props }) => "value" in props || "onSelect" in props);
+  const items = React.Children.toArray(children).filter(
```
Comment (akdetrick) [inline src/Select/index.js:70]: formatting only

File: src/Select/index.js [hunk 2]
```diff
@@ -66,7 +67,9 @@ const Select = ({
...
+    ({ props }) => "value" in props || "onSelect" in props
+  );
```
Comment (akdetrick) [inline src/Select/index.js:72]: formatting only

File: src/Select/index.js [hunk 3]
```diff
@@ -102,50 +105,78 @@ const Select = ({
...
-            >
-              {hasSelectedItem &&
-                selectedItem.props.value === item.props.value && (
-                  <span className="narmi-icon-check fontSize--l fontWeight--bold" />
-                )}
-              {item}
-            </li>
-          ))}
-      </ul>
+      {renderLayer(
```
Comment (akdetrick) [inline src/Select/index.js:136]: `renderLayer` creates a portal

File: src/Select/index.js [hunk 4]
```diff
@@ -102,50 +105,78 @@ const Select = ({
...
     <div className="nds-select" data-testid={testId}>
       <DropdownTrigger
-        isOpen={isOpen}
+        isOpen={showMenu}
         labelText={label}
         displayValue={getSelectedItemDisplay(selectedItem)}
         labelProps={{ ...getLabelProps() }}
         errorText={errorText}
-        {...getToggleButtonProps()}
+        {...getToggleButtonProps(triggerProps)}
```
Comment (akdetrick) [inline src/Select/index.js:130]: This is how we combine the `react-laag` element props with the `downshift` prop getters.

All the prop getters like this from `downshift` take an argument; an object of additional props to apply.

**Review by xiexiewenting — APPROVED** [review]
🎉

**Pull request merged**